exchange/exchange.go

```diff
@@ -1300,7 +1314,25 @@ func buildStoredAuctionResponse(storedAuctionResponses map[string]json.RawMessag
 			//set imp id from request
 			for i := range seat.Bid {
 				seat.Bid[i].ImpID = impId
-				bidsToAdd = append(bidsToAdd, &entities.PbsOrtbBid{Bid: &seat.Bid[i]})
+				mType := seat.Bid[i].MType
+				var bidType openrtb_ext.BidType
+				if mType > 0 {
+					if mType <= 4 {
+						//1 = Banner, 2 = Video,  3 = Audio, 4 = Native
+						bidType = openrtb_ext.BidTypes()[mType-1]
+					} else {
+						return nil, nil, nil, &errortypes.BadServerResponse{
+							Message: fmt.Sprintf("Failed to parse bid mType for impression \"%s\"", seat.Bid[i].ImpID),
+						}
+					}
```

Comment:
This code assumes the order of enums in PBS matches that of OpenRTB 3.0. This mapping is intentional, but subject to change in the future on either side. IMHO it would be better to be explicit:

```
if mType > 0 {
	switch mType {
	case openrtb2.MarkupBanner:
		bidType = openrtb_ext.BidTypeBanner
	case openrtb2.MarkupVideo:
		bidType = openrtb_ext.BidTypeVideo
	case openrtb2.MarkupAudio:
		bidType = openrtb_ext.BidTypeAudio
	case openrtb2.MarkupNative:
		bidType = openrtb_ext.BidTypeNative
	default:
		return nil, nil, nil, &errortypes.BadServerResponse{
			Message: fmt.Sprintf("Failed to parse bid mType for impression \"%s\"", seat.Bid[i].ImpID),
		}
	}
```

Reply:
Oh I see what you meant. Added!